aleph/views/mappings_api.py

```diff
@@ -0,0 +1,123 @@
+import logging
+
+from flask import Blueprint, request
+# from followthemoney import model
+# from followthemoney.exc import InvalidMapping
+from werkzeug.exceptions import BadRequest
+
+from aleph.core import db, archive
+from aleph.model import Mapping
+from aleph.search import QueryParser, DatabaseQueryResult
+from aleph.views.serializers import MappingSerializer, first
+from aleph.views.util import get_db_collection, get_index_entity, parse_request  # noqa
+from aleph.views.context import enable_cache
+from aleph.views.forms import MappingSchema
+from aleph.views.util import require, obj_or_404, get_session_id
+from aleph.queues import queue_task, OP_BULKLOAD
+from aleph.logic.collections import refresh_collection
+
+blueprint = Blueprint('mappings_api', __name__)
+log = logging.getLogger(__name__)
+
+
+@blueprint.route('/api/2/collections/<int:collection_id>/mappings', methods=['GET'])  # noqa
+def index_by_collection(collection_id):
+    collection = get_db_collection(collection_id)
+    parser = QueryParser(request.args, request.authz)
+    q = Mapping.by_collection(collection.id)
+    if 'table' in parser.filters:
+        q = Mapping.by_table(parser.filters['table'], q)
+    result = DatabaseQueryResult(request, q, parser=parser)
+    return MappingSerializer.jsonify_result(result)
+
+
+def _load_query():
+    try:
+        query = request.json.get('mapping_query', '{}')
+        # TODO: validate query
+    except Exception as ex:
+        raise BadRequest(str(ex))
+    return query
+
+
+@blueprint.route('/api/2/mappings', methods=['GET'])
+def index():
+    require(request.authz.logged_in)
+    parser = QueryParser(request.args, request.authz)
+    q = obj_or_404(Mapping.by_role_id(request.authz.id))
+    result = DatabaseQueryResult(request, q, parser=parser)
+    return MappingSerializer.jsonify_result(result)
+
+
+@blueprint.route('/api/2/mappings', methods=['POST', 'PUT'])
+def create():
```

Comment:
All of these actions need to authenticate on the collection. I'm wondering if it wouldn't be a good idea to prefix the API with `collections/xx` for that reason alone? At the moment, the security on this makes no sense.